docgen/src/gettingstarted.md

```diff
@@ -6,49 +6,323 @@ nav_groups:
 
-const App = () =>
+var App = () =>
   <InstantSearch
-    appId="latency"
-    apiKey="6be0576ff61c053d5f9a3225e2a90f76"
-    indexName="movies"
+    appId={appId}
+    apiKey={apiKey}
+    indexName={indexName}
   >
-    <div>
-      <SearchBox />
+   // The instantsearch components will go here
+  </InstantSearch>
+
+// Needed only if you're js app doesn't do it already.
+// Create-react-app does it for you ;)
+React.render(<App />, document.querySelector('#app'));
+```
+
+As you can see, there are a few mandatory props. Those props are the
+credentials of your application in Algolia. They can be found in your Algolia
+dasboard.
+
+Congratulations, your application is now connected to Algolia. This wrapper
+will provide a context so that the instantsearch components you add inside
+can interact with the search.
+
+In this section we've seen:
+ - how to connect a part of a react application to Algolia
+ - configure your credentials with react-instantsearch
+
+ > To get more *under the hood* information about the `InstantSearch` wrapper
+ > component, we have [full documentation](InstantSearch.html) in the widget
+ > section.
+
+## Display the results
+
+The core of a search is to display results. By default, react-instantsearch
+will do a query at the start of the page, because the relevance from the
+Algolia engine will always sort your records even though they are not
+filtered yet.
+
+Therefore the next widget we want to use is the Hits widget. This widget will
+display all the results returned by Algolia, and it will update when there
+are new results.
+
+And while we're doing that, let's create a new component for the search:
+
+```javascript
+// First, we need to add the Hits component to our import
+import {InstantSearch, Hits, SearchBox} from 'react-instantsearch';
+
+// ... We skip the rest of the code here
+
+function Search() {
+  return (
+    <div className='container'>
       <Hits />
     </div>
-  </InstantSearch>;
-};
+  );
+}
+```
+
+We can then call this component in the render of App:
+
+```javascript
+var App = () =>
+  <InstantSearch
+    appId={appId}
+    apiKey={apiKey}
+    indexName={indexName}
+  >
+   <Search/>
+  </InstantSearch>
+```
+
+You should now be able to see the results... in a very raw way. This
+view lets you inspect the values that are retrieve from Algolia, in order
+to build your custom view. In react-instantsearch, custom views are built
+using your own components that are linked to InstantSearch using connectors.
+
+Connectors are functions that will enhance your component by adding them
+custom search related logic, as well as providing them an API to read
+and write the search state. The connectors will have the same name as
+their connected component counterpart.
+
+Here we're going to use the Hits connector, since we want to implement
+a custom rendering for a Hits widget.
+
+ > *TODO*: provide a link to the Hits connector documentation
+
+```javascript
```

Comment:
Here, there's no practical need (but for teaching connectors) to talk about connectors, you can use `<Hits itemComponent={} />`

Should we speak about connectors in the getting started? I feel this is too detailed, the need for connectors arise as soon as you want to control the full rendering. But in the getting started I would stick to just discovering the big picture and get a running example in 5 minutes, no more.

While still for example adding "This RefinementList renders a list of links, you could also completely control the rendering and the binding of DOM events for refinements, have a look at our connectors guide".